bench/workgen/workgen.cxx

```diff
@@ -137,6 +137,102 @@ static void *thread_workload(void *arg) {
     return (NULL);
 }
 
+static void *thread_idle_table_cycle_worload(void *arg) {
+
+    WorkloadRunnerConnection *runnerConnection = (WorkloadRunnerConnection *) arg;
+    WorkloadRunner *runner = runnerConnection->runner;
+    WT_CONNECTION *connection = runnerConnection->connection;
+
+    try {
+        runner->start_table_idle_cycle(connection);
+    } catch (WorkgenException &wge) {
+        std::cerr << "Exception while create/drop tables." << std::endl;
+    }
+
+    return (NULL);
+}
+
+int WorkloadRunner::check_timing(const char *name, uint64_t start, uint64_t *stop) {
+    uint64_t last_interval;
+    int msg_err;
+    const char *str;
+    WorkloadOptions *options = &_workload->options;
+
+    msg_err = 0;
+
+    workgen_clock(stop);
+
+    last_interval = ns_to_sec(*stop - start);
+
+    if (last_interval > options->max_idle_table_cycle) {
+        if (options->max_idle_table_cycle_fatal) {
+            msg_err = ETIMEDOUT;
+            str = "ERROR";
+        } else {
+            str = "WARNING";
+        }
+        std::cerr << str << ": Cycling idle table failed because " << name << " took " << last_interval << " seconds which is longer than configured acceptable maximum of " << options->max_idle_table_cycle << std::endl;
+     }
+     return (msg_err);
+}
+
+int WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn) {
+
+    //std::string session_config = _workload->_threads[0].options.session_config;
+    int ret, cycle_count;
+    WT_SESSION *session;
+    WT_CURSOR *cursor;
+    char uri[BUF_SIZE];
+    uint64_t start, stop;
+
+    cycle_count = 0;
+    if (ret = conn->open_session(conn, NULL, NULL, &session) != 0) {
+        THROW ("Error Opening a Session.");
+    }
+
+    for (cycle_count = 0 ; !stopping ; ++cycle_count) {
+        sprintf (uri, "table:test_cycle%04d", cycle_count);
+
+        workgen_clock(&start);
+
+        /* Create a table. */
+        if ((ret = session->create(session, uri, "key_format=S,value_format=S")) != 0) {
+            if (ret == EBUSY)
+                continue;
+            THROW ("Table create failed in start_table_idle_cycle.");
+        }
+        if ((ret = check_timing("CREATE", start, &stop)) != 0)
+            THROW_ERRNO (ret, "WT_SESSION->create timeout.");
```

Comment:
Other examples of `THROW` `THROW_ERRNO` in this function that need spacing fixed.